index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
Can you provide some insight on how to use the @sintax on a context where no transpiling occurs?

I'm trying to follow something like [this](https://addyosmani.com/blog/decorator-pattern/)

If the Proxy hijacks the property definition, I'm not sure on how an export can be combined with the known instantiation to expose the decorator method.